### PR TITLE
add Profile and Background pic

### PR DESCRIPTION
**Pull request by hsnice16**
### Now, leaving a reply, redux code, and search, you can do anything on Think Loud

In this PR, **534 additions 140 deletions** lines of code
- many files are just assets and backend related
- some files contain only 5 to 6 lines of code change

### Please review this PR and give your feedback

---

#### [Live Preview](https://think-loud-git-functionality-picture-hsnice16.vercel.app/home)

### CODE REVIEW

**Review by foolhardy21 — APPROVED** [review]
The feature of uploading profile pic is working good. Also I appreciate the use of `useMemo`.

File: src/components/dialog/edit-profile-dialog/EditProfileDialog.jsx [hunk 1]
```diff
@@ -78,6 +89,24 @@ export const EditProfileDialog = ({
     }
   };
 
+  const handleFileInputChange = (event) => {
+    const selectedFile = event.target.files[0];
+
+    if (selectedFile) {
+      const reader = new FileReader();
+
+      reader.addEventListener("load", () => {
```
Comment (nayyyhaa) [inline src/components/dialog/edit-profile-dialog/EditProfileDialog.jsx:98]: check if there's an alternative for addEventListener here.

File: src/components/dialog/edit-profile-dialog/EditProfileDialog.jsx [hunk 2]
```diff
@@ -208,6 +268,7 @@ export const EditProfileDialog = ({
 EditProfileDialog.propTypes = {
   profileData: PropTypes.shape({
     bio: PropTypes.string,
+    bgPic: PropTypes.string,
     email: PropTypes.string,
     password: PropTypes.string,
     username: PropTypes.string,
```
Comment (nayyyhaa) [inline src/components/dialog/edit-profile-dialog/EditProfileDialog.jsx:274]: Appreciation: good that you are using propTypes here.

File: src/components/left-side-navbar/LeftSideNavbar.jsx [hunk 3]
```diff
@@ -122,6 +120,9 @@ export const LeftSideNavbar = () => {
           styles.listText_active,
           styles.accountCircle_btnText
         )}
+        onClick={() => setOpenLogoutDialog(true)}
+        fullName={`${data?.firstName ?? ""} ${data?.lastName ?? ""}`}
```
Comment (nayyyhaa) [inline src/components/left-side-navbar/LeftSideNavbar.jsx:124]: appreciation: good use of `??` nullish coalescing

File: src/components/shared/broadcast-box-header/BroadcastBoxHeader.jsx [hunk 4]
```diff
@@ -3,11 +3,19 @@ import classNames from "classnames";
...
+  h2Text,
+  children,
+  className,
+  linkProps,
+}) => {
   return (
     <Box className={classNames(styles.header, className)}>
       <Box className={styles.name}>
-        <Typography component="h2">{h2Text}</Typography>
+        <Typography component="h2" {...linkProps}>
```
Comment (nayyyhaa) [inline src/components/shared/broadcast-box-header/BroadcastBoxHeader.jsx:16]: appreciation: wow great use of sending attr in props

**Review by nayyyhaa — APPROVED** [review]
Nice work Himanshu! Code is well written with use of ES6 convention + useMemo + propTypes + all edge cases covered. You can merge the changes.

**Review by vrx29 — APPROVED** [review]
Nice work. All functionality is working great and code is clean. You can merge it.

File: src/components/dialog/edit-profile-dialog/EditProfileDialog.jsx (hunk 1 above)
Comment (hsnice16) [reply src/components/dialog/edit-profile-dialog/EditProfileDialog.jsx:98]: yup! will look. And if I will find, will add in next PR.
Thanks for suggesting the change.

**Comment by hsnice16** [comment]
Thank you for review.

**Pull request merged**